network/src/interface/mod.rs

```diff
@@ -304,7 +304,7 @@ where
                 }
             }
             _ => {
-                warn!(
+                unexpected!(
```

Comment:
I think that this is probably not better than using a warn in this case. My main concern which the addition of new macros like this is, what does unexpected do? most folks know what panic and friends do as well as log and friends but since unexpected is new its hard to tell without looking at the macro definition itself to determine what's happening.